pkg/asset/machines/openstack/machinesets.go

```diff
@@ -41,13 +41,17 @@ func MachineSets(clusterID string, config *types.InstallConfig, pool *types.Mach
 		{
 			// The replica number is set to 3 by default when install-config does not have
 			// any Compute machine-pool, or when the Compute machine-pool does not have the
-			// `replicas` property. As a consequence, pool.Replicas is never nil
-			replicas := *pool.Replicas / numberOfFailureDomains
-			if int64(idx) < *pool.Replicas%numberOfFailureDomains {
-				replicas++
-			}
-			if replicas > maxInt32 {
-				return nil, fmt.Errorf("the number of requested worker replicas (%d) is too high. Each MachineSet can hold %d replicas; the install-config encodes for %d MachineSets, which gives us a replica number of %d", *pool.Replicas, maxInt32, numberOfFailureDomains, replicas)
+			// `replicas` property.
+			// However, external consumers of this func may not be so kind...
+			var replicas int64 = 0
+			if pool.Replicas != nil {
+				replicas = *pool.Replicas / numberOfFailureDomains
+				if int64(idx) < *pool.Replicas%numberOfFailureDomains {
+					replicas++
+				}
+				if replicas > maxInt32 {
+					return nil, fmt.Errorf("the number of requested worker replicas (%d) is too high. Each MachineSet can hold %d replicas; the install-config encodes for %d MachineSets, which gives us a replica number of %d", *pool.Replicas, maxInt32, numberOfFailureDomains, replicas)
+				}
```

Comment:
This code exercises different code paths in case `pool.Replicas` is `nil` or `&0`. Since both cases amount to the same effect, it would be good to have them share as much code as possible, so that tests on either side will catch potential bugs.

What do you think of this?

```suggestion
			var poolReplicas int64
			if pool.Replicas != nil {
				poolReplicas = *pool.Replicas
			}
			replicas := poolReplicas / numberOfFailureDomains
			if int64(idx) < poolReplicas%numberOfFailureDomains {
				replicas++
			}
			if replicas > maxInt32 {
				return nil, fmt.Errorf("the number of requested worker replicas (%d) is too high. Each MachineSet can hold %d replicas; the install-config encodes for %d MachineSets, which gives us a replica number of %d", *pool.Replicas, maxInt32, numberOfFailureDomains, replicas)
			}
```

Reply:
Fair point.

Your suggestion would also require swapping out the `*pool.Replicas` on L55.

I'm pushing a simpler fix, though. The only caveat is that it mutates the `pool`, but that seems acceptable to me. Thoughts?

Reply:
I like your new solution! I don't think your mutation sin will have consequences, but if it does, it's because we made some bad assumptions somewhere else.

Reply:
One last nit: I think you can move your not-nil-enforcer code outside the `for range machineset` loop, somewhere around line 34.